atlasdb-jepsen-tests/src/jepsen/atlasdb/lock.clj

```diff
@@ -0,0 +1,108 @@
+   function will return an invalid object: you should call 'setup' on the returned object to get a valid one.
+  "
+  [lock-service, client-name, token-store]
+  (reify client/Client
+    (setup!
+      [this test node]
+      "Factory that returns an object implementing client/Client"
+        (create-client
+          (LockClient/create '("n1" "n2" "n3" "n4" "n5"))
+          (name node)
+          (create-token-store)))
+
+    (invoke!
+      [this test op]
+      "Run an operation on our client"
+      (timeout (* 30 1000)
+          (assoc op :type :fail :error :timeout)
+          (try
+            (let [lock-name (:value op)
+                  token-store-key (str client-name lock-name)]
+              (case (:f op)
+                :lock
+                  (let [response (LockClient/lock lock-service client-name lock-name)]
+                    (do (replace-token token-store token-store-key response)
+                        (assoc-ok-value op response [client-name lock-name])))
+                :unlock
+                  (let [token (@token-store token-store-key)
+                        response (LockClient/unlock lock-service token)]
+                    (do (replace-token token-store token-store-key nil)
+                        (assoc-ok-value op response token)))
+                :refresh
+                  (let [token (@token-store token-store-key)
+                        response (LockClient/refresh lock-service token)]
+                    (do (replace-token token-store token-store-key token)
+                        (assoc-ok-value op response token)))))
+          (catch Exception e
+            (assoc op :type :fail :error (.toString e))))))
+
+    (teardown! [_ test])))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; How to check the validity of a run of the Jepsen test. Hard-coded to succeed for now.
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(def checker
+  (reify checker/Checker
+    (check [this test model history opts]
+      {:valid? true})))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; How to generate test events. We randomly mix refreshes, locks, and unlocks.
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+(def generator
+  (reify gen/Generator
+    (op [generator test process]
+      (let [random-lock-name (rand-nth lock-names)]
+        (condp < (rand)
+          ;; 70% chance of a refresh
+          0.30 {:type :invoke   :f :refresh   :value random-lock-name}
+          ;; 15% chance of a lock
+          0.15 {:type :invoke   :f :lock      :value random-lock-name}
+          ;; 15% chance of an unlock
+               {:type :invoke   :f :unlock    :value random-lock-name})))))
```

Comment:
What this does is it gets a random real (actually float) from 0 to 1, and then checks if it is less than 0.3, then less than 0.15.

nit: I think writing it in the other way as a cumulative distribution would be clearer.

```
(condp > (rand)
  0.70 ... ;;refresh
  0.85 ... ;;lock
       ... ;;unlock
```

Matter of style though, so feel free to push back.